rust/src/lib.rs

```diff
@@ -739,15 +783,22 @@ pub unsafe extern "C" fn zcashlc_get_next_available_address(
     db_data_len: usize,
     account_uuid_bytes: *const u8,
     network_id: u32,
+    receiver_flags: u8,
 ) -> *mut c_char {
     let res = catch_panic(|| {
         let network = parse_network(network_id)?;
         let mut db_data = unsafe { wallet_db(db_data, db_data_len, network)? };
         let account_uuid = account_uuid_from_bytes(account_uuid_bytes)?;
+        let receiver_flags = ReceiverFlags::from_bits(receiver_flags)
+            .ok_or_else(|| anyhow!("Invalid unified address receiver flags {}", receiver_flags))?;
+        let address_request = receiver_flags.to_address_request().map_err(|_| {
+            anyhow!(
+                "Could not generate a valid unified address for flags {}",
+                receiver_flags.bits()
+            )
+        })?;
 
-        match db_data
-            .get_next_available_address(account_uuid, UnifiedAddressRequest::AllAvailableKeys)
```

Comment:
I confirmed that the previous value can be expressed with the new API, as it happens to only use `Omit` and `Require`, and the choice of these is fixed in our compiled FFI binary.